### PR TITLE
fix: report all indeterminable anchors

### PR DESCRIPTION
**Pull request by patrickhulce**
R: @ebidel 

Changes rel noopener audit to warn about all potentially violating anchor tags even when URL is malformed.

Addresses #1345

### CODE REVIEW

File: lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js [hunk 1]
```diff
@@ -49,12 +49,23 @@ class ExternalAnchorsUseRelNoopenerAudit extends Audit {
...
     // https://github.com/GoogleChrome/lighthouse/issues/1233
+    // https://github.com/GoogleChrome/lighthouse/issues/1345
     const failingAnchors = artifacts.AnchorsWithNoRelNoopener.usages
-      .filter(anchor => anchor.href === '' || new URL(anchor.href).host !== pageHost)
+      .filter(anchor => {
+        try {
+          return new URL(anchor.href).host !== pageHost;
+        } catch (err) {
+          debugString = 'Lighthouse was unable to determine the destination ' +
+              'of some anchor tags. Remove the href attribute if they are not ' +
```
Comment (ebidel) [inline lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: One of the things Brendan discovered in the last PR for this audit was that many SPAs don't include an href attr and use click handlers instead. You catch will include those cases, but maybe we should modify the debug string to say so. E.g. the missing href is intentional in their app, but we're still saying remove it.

Comment (ebidel) [inline lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: IOW might want to to still one off the href === "" case

Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: ah I didn't realize `anchor.href === ''` would be true even if it didn't have an href at all. This seems like an ambiguous case then. There'd be a false positive most of the time if we warn for all missing `href`s but it's feasible they still send externally.

**Comment by ebidel** [comment]
Yea. The underlying js property always gets set. Crazy HTML.

We'll still only warn if the a also has target="_blank" so it may not be
all that noisy.

File: lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js]: True, SG then I'll add it back in.

**Pull request merged**